src/test/scala/chiselTests/Assert.scala

```diff
@@ -8,12 +8,52 @@ import Chisel.testers.BasicTester
 
 class FailingAssertTester() extends BasicTester {
   assert(Bool(false))
-  stop()
+  // Wait to come out of reset
+  val (_, done) = Counter(!reset, 2)
+  when (done) {
+    stop()
+  }
 }
 
 class SucceedingAssertTester() extends BasicTester {
   assert(Bool(true))
-  stop()
+  // Wait to come out of reset
+  val (_, done) = Counter(!reset, 2)
+  when (done) {
+    stop()
+  }
+}
+
+class PipelinedResetModule extends Module {
+  val io = new Bundle { }
+  val a = Reg(init = UInt(0xbeef))
+  val b = Reg(init = UInt(0xbeef))
+
+  assert(a === b)
+}
+
+class PipelinedResetTester(depth: Int) extends BasicTester {
+  // Run long enough for:
+  //   1. reset assert to propogate (depth)
+  //   2. reset deassert to propogate (depth)
+  //   3. a few cycles past
+  val simLength = depth * 3
+
+  val module = Module(new PipelinedResetModule)
+  val testReset = Reg(init = Bool(true))
+  val pipeReset = Pipe(Bool(true), testReset, depth)
+
+  module.reset := pipeReset.bits
+  module.globalReset := testReset || reset
+  // Without (|| reset), we never come out of reset
```

Comment:
For the top-level module, shouldn't globalReset implicitly equal reset?  as opposed to exposing it as a pin the user has to supply.